seg_lapa/config_parse/checkpoint_conf.py

```diff
@@ -0,0 +1,41 @@
+from abc import ABC, abstractmethod
+from typing import Optional
+
+import pytorch_lightning as pl
+from omegaconf import DictConfig
+from pydantic.dataclasses import dataclass
+from pytorch_lightning.callbacks import ModelCheckpoint
+
+from seg_lapa.config_parse.conf_utils import asdict_filtered, validate_config_group_generic
+from seg_lapa.utils.path_check import get_project_root
+
+
+@dataclass
+class CheckpointConf(ABC):
+    name: str
+
+    @abstractmethod
+    def get_checkpoint_callback(self, *args) -> pl.callbacks:
+        pass
+
+
+@dataclass
+class CheckpointConfig(CheckpointConf):
+    dirpath: Optional[str]
+
+    def get_checkpoint_callback(self, cfg: DictConfig, run_id) -> pl.callbacks:
+        if self.dirpath is None:
+            self.dirpath = str(get_project_root())+'/logs/'+run_id
+
+        checkpoint_callback = ModelCheckpoint(**asdict_filtered(self))
```

Comment:
This saves the checkpoints for a run in the given directory, yes?

What other logs are generated by lightning? The trainer has a root_dir parameter. That should also be set
That param will control where the pytorch-lightning-logs dir is created. But will probably still create subdirs called  version0, version1, etc. We'd ideally like the change the naming scheme of those subdirs to use wandb runid instead and save the checkpoints there as well.

Reply:
This will create the run_id folder inside logs folder, and save checkpoints inside the run_id folder. This is expected behavior, right?
Will check what other logs are generated by lightning. 
Wandb logs are also generated. Should I disable it?